lightning/src/chain/mod.rs

```diff
@@ -57,6 +57,81 @@ pub trait Listen {
 	fn block_disconnected(&self, header: &BlockHeader, height: u32);
 }
 
+/// The `Confirm` trait is used to notify when transactions have been confirmed on chain or
+/// unconfirmed during a chain reorganization.
+///
+/// Clients should use this interface instead of [`Listen`] if they receive transaction data only
```

Comment:
I think the distinction as to when to use this API vs Listen is more about if you get chain data in a transaction-oriented API vs a blockchain-oriented API - ie do you receive notifications of each new header with relevant transaction data, or do you receive simply notifications of transaction-level state changes, with header data separated out.

Reply:
Right. I do kinda agree with you that "transaction-oriented" isn't really clear for a user, but I'm not sure what is better, and I do think it captures exactly what we mean. I wonder if adding more words to describe it would help or if that just makes it more confusing. Maybe it can be resolved by just mentioning "Clients which fetch the full header chain" in the docs for `Listen`?

Reply:
Good idea -- expanded `Listen` docs to mention the full header chain. That said, it seems the only difference between the two APIs in this regard is that `Listen` requires the full header chain whereas `Confirm` is simply ok with skipping headers . Unless we want to further relax how `Confirm` docs describe when to call `best_block_updated`. Currently, it just says "whenever a new chain tip becomes available".

Reply:
As long as we don't expand the docs around `best_block_updated` to include the reorg case, the reorg handling is a pretty big difference, too, I think. https://github.com/rust-bitcoin/rust-lightning/pull/889#discussion_r617773681